[wip] 148034075 cw post foods v2

josh-works opened the pull request:
we're struggling hard with a POST request. Notes in the files.

josh-works committed:
[wip] setup

caward12 committed:
[WIP] post foods

caward12 committed:
[WIP] post to foods

josh-works commented on test/server-test.js:
```diff
@@ -87,6 +87,31 @@ describe('Server', function(){
       })
     })
+
+    describe("POST /foods", function(){
```
here's our POST test. It's technically passing, but we're asserting the response from the server is hardcoded to the happy path.

We're working on adding the sad path, but this is a struggle.

josh-works commented on test/server-test.js:
```diff
@@ -87,6 +87,31 @@ describe('Server', function(){
+          assert.equal(response.statusCode, 201)
+          // assert.equal(Foods.countFoods(), 2)
+          assert.equal(parsedFood.status, "success")
+          assert.equal(parsedFood.message, 'Inserted one food')
```
two wimpy assertions, pointing over to... one moment.

josh-works commented on server.js:
```diff
@@ -1,17 +1,36 @@
     response.json(data.rows)
   })
 })
+// POST foods
```
post request path, I believe.

DBULL7 replied:
It is. is Foods an object you're importing?

caward12 committed:
[WIP] add sad path to POST foods

caward12 committed:
[WIP] put foods

josh-works commented on server.js:
```diff
@@ -39,13 +39,23 @@ app.post('/api/v1/foods', function(request, response) {
 })
 
+// PUT foods/:id
+app.put('/api/vi/foods/:id', function (request, response){
```
we're not able to get this function to run (it seems) from the test file.

The `console.log()` on line 52 isn't being hit.

josh-works commented on test/server-test.js:
```diff
@@ -123,6 +123,30 @@ describe('Server', function(){
+      it("can update existing food", function(done){
+        var food = {name: "cheese pizza"}
+
+        this.request.put("api/v1/foods/1", {form: food}, function(error, response){
```
this `put` request should trigger the console.log in the above file. It's not, for some reason.

Any ideas on how to get that console.log (in server.js:52) to run?

@DBULL7 ^^

caward12 committed:
put and post foods

[closes #148034075]
[closes #148033965]
testing happy and sad path post
testing happy path for put

caward12 committed:
delete foods

caward12 committed:
update test for put foods
